internal/bminventory/inventory.go

```diff
@@ -1647,12 +1647,14 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 	}
 
 	var err error
-	*machineCidr, err = network.CalculateMachineNetworkCIDR(apiVip, ingressVip, cluster.Hosts)
+	*machineCidr, err = network.CalculateMachineNetworkCIDR(apiVip, ingressVip, cluster.Hosts, false)
```

Comment:
We talked about not changing the current flow of the rest-api. we still want to guide the users not to make a mistake.
But maybe it can be done better this way? @avishayt @ori-amizur what do you think?
Users will be able to set API VIP and ingress VIP without setting machine network cidr, just need to make sure that the validation is clear, in this case it should be in pending because there are no hosts.

Reply:
as I consulted with @ori-amizur we can just perform the check if we have enough data, if not it will happen anyway before installation. But I can change that logic to happen only when cluster is crd so it will be more intuitive.

Reply:
I'm not against it, just making sure that everyone are synced.

Reply:
Come to think of it, I tend to agree with @filanov .  The current REST API behavior should not change.  So the behavior that checks the enough data is available is relevant only to non REST API